platform/linuxbsd/display_server_x11.cpp

```diff
@@ -1865,6 +1865,16 @@ bool DisplayServerX11::_window_fullscreen_check(WindowID p_window) const {
 	return retval;
 }
 
+void DisplayServerX11::_validate_fullscreen_on_map(WindowID p_window) {
+	const WindowData &wd = windows[p_window];
+	if (wd.fullscreen && !_window_fullscreen_check(p_window)) {
+		// Undo the failed fullscreen mode.
+		window_set_mode(WINDOW_MODE_WINDOWED, p_window);
+		// Try again now that we are mapped.
+		window_set_mode(WINDOW_MODE_FULLSCREEN, p_window);
```

Comment:
Not sure about toggling it back and forth, maybe it's better to directly set the flag to avoid any intermediate window size changes, like (not tested):
```c++
_set_wm_fullscreen(p_window, true);
```

Reply:
I have amended the PR with the requested changes. Testing suggests the change works fine. I also looked into your suggestion that `WINDOW_MODE_MINIMIZED` and `WINDOW_MODE_MAXIMIZED` will be affected by this bug; I can confirm that this is the case. Should this PR be amended to deal with those cases as well?

Reply:
Let's merge this one as is since it's ready (and I'm about to build 4.0 alpha 15), but feel free to open a PR with the other fixes.